obspy/io/sac/sactrace.py

```diff
@@ -1092,7 +1092,7 @@ def read(cls, source, headonly=False, ascii=False, byteorder=None,
                 val = _ut._clean_str(val, strip_whitespace=False)
                 if val.startswith(native_str('-12345')):
                     val = HD.SNULL
-                hs[i] = val
+                hs[i] = val.encode('ASCII', 'replace')
```

Comment:
Thanks for your comment. `hs[i]` was an array of bytes. So, the previous statement will make the implicit conversion which is equivalent to `hs[i] = val.encode('ASCII', 'strict')`. However, when header string is not encoded in 'ASCII', it causes a problem and thus raises an exemption. This PR was trying to let the non-ASCII characters pass the `read` function for now as '?' and leave the implementation of a 'encoding' flag for the later version as we discussed before.